### PR TITLE
Fix minor issues to run with OpenMP

### LINKED ISSUE #102: Fix simulations with Kokkos::OpenMP
- CMakeLists need minor updates when Kokkos is using OpenMP
- Add a critical section in the ArborX callback

### PR DESCRIPTION
**Pull request by uhetmaniuk**
Closes #102 

This branch was created to address minor issues when running with Kokkos + OpenMP.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,17 +125,26 @@ endif()
 # If trilinos is used, it comes with Kokkos so we don't need standalone
 if(NOT HAVE_TRILINOS AND HAVE_KOKKOS)
   if(HAVE_KOKKOS)
-    if(KOKKOS_ROOT_DIR)
+    if (Kokkos_ROOT)
```
Comment (nmm0) [inline CMakeLists.txt:128]: I'm sort of wondering if this extra branch is still needed. `target_link_libraries(nimble PUBLIC Kokkos::kokkos)` is in theory supposed to set OpenMP flags if kokkos is built with OpenMP and also set the include directories correctly

**Review by nmm0 — APPROVED** [review]
Looks good!

File: CMakeLists.txt (hunk 1 above)
Comment (uhetmaniuk) [reply CMakeLists.txt]: I removed a test that was always true.

I am not sure to understand your comment about OpenMP. I agree with you about the libraries and the include directories. 

I do not believe the CXXFLAGS will be propagated and will contain "-fopenmp". What do you think?

Comment (nmm0) [reply CMakeLists.txt]: If Kokkos is set up properly (which I think it is for more recent version of Kokkos), doing `target_link_libraries(nimble PUBLIC Kokkos::kokkos)` will also set "PUBLIC" compiler flags, such as fopenmp, etc to any target it's linked to. It's similar to how flags are propagated from the libnimble target to the various executables

Comment (uhetmaniuk) [reply CMakeLists.txt]: Fixed CMakeLists.txt

**Pull request merged**